website/static/js/accountSettings.js

```diff
@@ -0,0 +1,170 @@
+'use strict';
+
+var $ = require('jquery');
+var $osf = require('js/osfHelpers.js');
+var ko = require('knockout');
+var oop = require('js/oop.js');
+var Raven = require('raven-js');
+
+require('knockout.punches');
+ko.punches.enableAll();
+
+
+var userEmail = oop.defclass({
+    constructor: function(data) {
+        this.address = ko.observable();
+        this.isConfirmed = ko.observable();
+        this.isPrimary = ko.observable();
+        if (typeof(data) !== 'undefined') {
+            this.unserialize(data);
+        }
+    },
+    serialize: function() {
+        return {
+            address: this.address(),
+            primary: this.isPrimary(),
+            confirmed: this.isConfirmed()
+        };
+    },
+    unserialize: function(data) {
+        this.address(data.address);
+        this.isPrimary(data.primary || false);
+        this.isConfirmed(data.confirmed || false);
+    }
+});
+
+
+var userProfile = oop.defclass({
+    constructor: function(urls) {
+        this.urls = urls;
+
+        // TODO: Pass these in instead!
+        this.urls = {
```

Comment:
If you separate the CRUD operations into the `ProfileClient` suggested in my previous comment, you won't need to pass these in; they can just be internal data for the `ProfileClient`.